lib/routes/index.js

```diff
@@ -25,6 +25,7 @@ module.exports = function (
   var idp = require('./idp')(log, serverPublicKeys)
   var checkPassword = require('./utils/password_check')(log, config, Password, customs, db)
   var push = require('../push')(log, db)
+  var metricsContext = require('../metrics/context')(log, config)
```

Comment:
We're instantiating this module both here and in `log.js`, will that create two different memcached client objects with separate connections etc?  It may be worth trying to use a single shared instance to reduce resource contention.